hazelcast/src/main/java/com/hazelcast/instance/impl/Node.java

```diff
@@ -817,17 +817,25 @@ Joiner createJoiner() {
 
         if (properties.getBoolean(DISCOVERY_SPI_ENABLED) || isAnyAliasedConfigEnabled(join)) {
             //TODO: Auto-Upgrade Multicast+AWS configuration!
-            logger.info("Activating Discovery SPI Joiner");
+            if (logger.isFineEnabled()) {
```

Comment:
Knowing which joiner is being used is pretty useful.

Reply:
Yeah, I thought that this one might be controversial. Maybe we could just adjust the logging messages to be actually more user friendly? Right now it looks like a regular debugging message. As a new user, I don't know what the Joiner is (I shouldn't know, it's internal) and also I might better understand other word than "Activating", sounds like a robot.

How about I would change the log lines to be more user friendly like: 
```
Using Discovery API
Using multicast discovery
Using TCP/IP discovery
Using AWS discovery
```
and keep it on `info` level?

Reply:
Fine with me. Can't we get the discovery interface have a proper toString? So that it also works for custom discovery mechanisms.

Reply:
I have updated the PR. I didn't go with the road of providing `toString` method as this is going a bit deeper and I have no idea about the consequences.